pkg/arvo/lib/poke-proxy-hook.hoon

```diff
@@ -0,0 +1,219 @@
+/+  resource, default-agent, dbug
```

Comment:
I agree with you that it seems the best approach here is to make this its own app rather than a library. Seems unnecessary for us to make individual versions of this app for each use case, given how little is different between groups and graphs so far. Making this an individual app would also improve client-side ergonomics, as we can keep track of pending pokes via a singular subscription.

Something I'm curious about is the idea of chaining various pokes and then "inverting them" if one fails. We handle this via threads currently, but I see potential in this architecture to do something similar. For instance, sending a group-store join that "fails" and removes the group if any of the subscriptions to contacts, metadata, or group is nacked. That is a "pending complex transaction" which is more complicated than a chat message, but would be extraordinarily useful for communicating internal state to users.

Reply:
Sadly, writing this as an app is thwarted by the cage of a cage conversion problem. I guess we could just emit the fact as json and do the conversion in the agent with some json conversions.